quinn-proto/src/stream.rs

```diff
@@ -23,11 +25,114 @@ pub struct Streams {
+        let mut streams = FnvHashMap::default();
+        for i in 0..max_remote_uni {
+            streams.insert(
+                StreamId::new(!side, Directionality::Uni, u64::from(i)),
+                Recv::new().into(),
+            );
+        }
+        for i in 0..max_remote_bi {
+            streams.insert(
+                StreamId::new(!side, Directionality::Bi, i as u64),
+                Stream::new_bi(),
+            );
+        }
+        Self {
+            streams,
+            next_uni: 0,
+            next_bi: 0,
+            max_uni: 0,
+            max_bi: 0,
+            max_remote_uni,
+            max_remote_bi,
+            next_remote_uni: 0,
+            next_remote_bi: 0,
+            next_reported_remote_uni: 0,
+            next_reported_remote_bi: 0,
+        }
+    }
+
+    pub fn open(&mut self, side: Side, direction: Directionality) -> Option<StreamId> {
+        let (id, stream) = match direction {
+            Directionality::Uni if self.next_uni < self.max_uni => {
+                self.next_uni += 1;
+                (
+                    StreamId::new(side, direction, self.next_uni - 1),
+                    Send::new().into(),
+                )
+            }
+            Directionality::Bi if self.next_bi < self.max_bi => {
+                self.next_bi += 1;
+                (
+                    StreamId::new(side, direction, self.next_bi - 1),
+                    Stream::new_bi(),
+                )
+            }
+            _ => {
+                return None;
+            }
+        };
+        assert!(self.streams.insert(id, stream).is_none());
+        Some(id)
+    }
+
+    pub fn alloc_remote_stream(&mut self, side: Side, ty: Directionality) {
+        let (id, stream) = match ty {
+            Directionality::Bi => {
+                self.max_remote_bi += 1;
+                (
+                    StreamId::new(!side, Directionality::Bi, self.max_remote_bi - 1),
+                    Stream::new_bi(),
+                )
+            }
+            Directionality::Uni => {
+                self.max_remote_uni += 1;
+                (
+                    StreamId::new(!side, Directionality::Uni, self.max_remote_uni - 1),
+                    Recv::new().into(),
+                )
+            }
+        };
+        self.streams.insert(id, stream);
+    }
+
+    pub fn accept(&mut self, side: Side) -> Option<StreamId> {
+        if self.next_remote_uni > self.next_reported_remote_uni {
+            let x = self.next_reported_remote_uni;
+            self.next_reported_remote_uni = x + 1;
+            Some(StreamId::new(!side, Directionality::Uni, x))
+        } else if self.next_remote_bi > self.next_reported_remote_bi {
+            let x = self.next_reported_remote_bi;
+            self.next_reported_remote_bi = x + 1;
+            Some(StreamId::new(!side, Directionality::Bi, x))
+        } else {
+            None
+        }
+    }
+
+    pub fn reset(&mut self, side: Side) {
```

Comment:
```suggestion
    pub fn zero_rtt_rejected(&mut self, side: Side) {
```
(or something else) to avoid confusion with the unrelated `RESET_STREAM` operation (and adjust the comment below to match)